packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,722 @@
+import * as ecr from '@aws-cdk/aws-ecr';
+import { DockerImageAsset } from '@aws-cdk/aws-ecr-assets';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnService } from './';
+
+/**
+ * The image repository types
+ */
+export enum ImageRepositoryType {
+  /**
+   * Amazon ECR Public
+   */
+  ECR_PUBLIC = 'ECR_PUBLIC',
+
+  /**
+   * Amazon ECR
+   */
+  ECR = 'ECR',
+}
+
+/**
+ * The number of CPU units reserved for each instance of your App Runner service.
+ */
+export class Cpu {
+  /**
+   * 1 vCPU
+   */
+  public static readonly ONE_VCPU = Cpu.of('1 vCPU')
+
+  /**
+   * 2 vCPU
+   */
+  public static readonly TWO_VCPU = Cpu.of('2 vCPU')
+
+  /**
+   * Custom CPU unit
+   *
+   * @param unit custom CPU unit
+   */
+  public static of(unit: string) { return new Cpu(unit); }
+
+  /**
+   *
+   * @param unit The unit of CPU.
+   */
+  private constructor(public readonly unit: string) {}
+}
+
+
+/**
+ * The amount of memory reserved for each instance of your App Runner service.
+ */
+export class Memory {
+  /**
+   * 2 GB
+   */
+  public static readonly TWO_GB = Memory.of('2 GB')
+
+  /**
+   * 3 GB
+   */
+  public static readonly THREE_GB = Memory.of('3 GB')
+
+  /**
+   * 4 GB
+   */
+  public static readonly FOUR_GB = Memory.of('4 GB')
+
+  /**
+   * Custom Memory unit
+   *
+   * @param unit custom Memory unit
+   */
+  public static of(unit: string) { return new Memory(unit); }
```

Comment:
Is there a documentation page somewhere which tells you what values are valid? If so, would be great to link it there (even if the only options _at this time_ are those for which you have constants above...).

Reply:
Same as CPU. From the UI experience, when we select `1 vCPU` we can select memory with 2/3/4 GB while `2 vCPU` only `4 GB` is available. I am adding a note in the doc string.